src/twisted/web/http.py

```diff
@@ -377,6 +379,75 @@ def parseContentRange(header):
 
 
 
+class _IDeprecatedHTTPChannelToRequestInterface(Interface):
+    """
+    The interface L{HTTPChannel} expects of L{Request}.
```

Comment:
Maybe "uses" instead of "expects".  The full interface of the `Request` type is wider - but `HTTPChannel` doesn't use the rest of it.

Also, a reference in the other direction might be handy.  That is, in the `HTTPChannel` documentation for its own `requestFactory` attribute - saying it must create providers of this interface.